src/GraphicsDrawer.cpp

```diff
@@ -321,6 +321,28 @@ void GraphicsDrawer::_updateScreenCoordsViewport(const FrameBuffer * _pBuffer) c
 	gSP.changed |= CHANGED_VIEWPORT;
 }
 
+void GraphicsDrawer::_updateScreenCoordsViewport2(const FrameBuffer* _pBuffer) const
+{
+	s32 X, Y, WIDTH, HEIGHT;
+	const s32 scale = config.frameBufferEmulation.nativeResFactor;
+	const FrameBuffer* pCurrentBuffer = _pBuffer != nullptr ? _pBuffer : frameBufferList().getCurrent();
+	f32 scaleX, scaleY;
+	if (pCurrentBuffer != nullptr) {
+		scaleX = scaleY = pCurrentBuffer->m_scale;
+	}
+	else {
+		scaleX = dwnd().getScaleX();
+		scaleY = dwnd().getScaleY();
+	}
+	X = roundup(-1024.0f, scaleX);
+	Y = roundup(-1024.0f, scaleY);
+	WIDTH = roundup(2048.0f, scaleX);
+	HEIGHT = roundup(2048.0f, scaleY);
```

Comment:
It would be good to define constants for these numbers, for example
const f32 viewportOffsetXf = -1024.0f;
const s32 viewportOffsetX = -1024;
const f32 viewportWidthf = 2048.0f;
const s32 viewportWidth = 2048;

Reply:
Shall I add them in GraphicsDrawer.h?

Reply:
I've got another idea. If you change signature of _updateScreenCoordsViewport2 to void `_updateScreenCoordsViewport2(u32 _scale = 0) const;`
then I can call it from texrect drawer with scale 1, so these constants need not be exposed.
If _scale is zero, the function should get the scale as it does it now when _pBuffer = nullptr.

Reply:
And _updateScreenCoordsViewport2  is better to rename to _updateViewport because _updateViewport() is unused now.

Reply:
> If you change signature of _updateScreenCoordsViewport2 to 
```void _updateScreenCoordsViewport2(f32 _scale = 0.0f) const;```
 then I can call it from texrect drawer with scale 1.0f

Is it fine like it is now?